twine/commands/register.py

```diff
@@ -18,122 +18,45 @@
 
     resp.raise_for_status()
 
 
 def main(args):
     parser = argparse.ArgumentParser(prog="twine register")
-    parser.add_argument(
-        "-r", "--repository",
-        action=utils.EnvironmentDefault,
-        env="TWINE_REPOSITORY",
-        default=None,
-        help="The repository (package index) to register the package to. "
-             "Should be a section in the config file. (Can also be set "
-             "via %(env)s environment variable.) "
-             "Initial package registration no longer necessary on pypi.org: "
-             "https://packaging.python.org/guides/migrating-to-pypi-org/",
-    )
-    parser.add_argument(
-        "--repository-url",
-        action=utils.EnvironmentDefault,
-        env="TWINE_REPOSITORY_URL",
-        default=None,
-        required=False,
-        help="The repository (package index) URL to register the package to. "
-             "This overrides --repository. "
-             "(Can also be set via %(env)s environment variable.)"
-    )
-    parser.add_argument(
-        "-u", "--username",
-        action=utils.EnvironmentDefault,
-        env="TWINE_USERNAME",
-        required=False, help="The username to authenticate to the repository "
-                             "(package index) as. (Can also be set via "
-                             "%(env)s environment variable.)",
-    )
-    parser.add_argument(
-        "-p", "--password",
-        action=utils.EnvironmentDefault,
-        env="TWINE_PASSWORD",
-        required=False, help="The password to authenticate to the repository "
-                             "(package index) with. (Can also be set via "
-                             "%(env)s environment variable.)",
-    )
-    parser.add_argument(
-        "-c", "--comment",
-        help="The comment to include with the distribution file.",
-    )
-    parser.add_argument(
-        "--config-file",
-        default="~/.pypirc",
-        help="The .pypirc config file to use.",
-    )
-    parser.add_argument(
-        "--cert",
-        action=utils.EnvironmentDefault,
-        env="TWINE_CERT",
-        default=None,
-        required=False,
-        metavar="path",
-        help="Path to alternate CA bundle (can also be set via %(env)s "
-             "environment variable).",
-    )
-    parser.add_argument(
-        "--client-cert",
-        metavar="path",
-        help="Path to SSL client certificate, a single file containing the "
-             "private key and the certificate in PEM format.",
-    )
+    settings.Settings.register_argparse_arguments(parser)
     parser.add_argument(
         "package",
         metavar="package",
         help="File from which we read the package metadata.",
     )
 
     args = parser.parse_args(args)
+    register_settings = settings.Settings.from_argparse(args)
 
     # Call the register function with the args from the command line
-    register(**vars(args))
+    register(register_settings, args.package)
```

Comment:
Our Settings class distinctly only cares about what it cares about. Extra arguments need to be handled separately by the user of the object. This is something I could see us revising to allow for a unified usage.